### PR TITLE
notification icons

### PR DESCRIPTION
**Pull request by wunter8**

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -75,6 +75,7 @@ var (
 	fileRegex        = regexp.MustCompile(`^/file/([-_A-Za-z0-9]{1,64})(?:\.[A-Za-z0-9]{1,16})?$`)
 	disallowedTopics = []string{"docs", "static", "file", "app", "settings"} // If updated, also update in Android app
 	attachURLRegex   = regexp.MustCompile(`^https?://`)
+	iconURLRegex     = regexp.MustCompile(`^https?://`)
```
Comment (binwiederhier) [inline server/server.go:78]: i'd just rename this to urlRegex or something

File: client/client.go [hunk 2]
```diff
@@ -65,6 +66,13 @@ type Attachment struct {
 	Owner   string `json:"-"` // IP address of uploader, used for rate limiting
 }
 
+// Icon represents a message icon
+type Icon struct {
+	URL  string `json:"url"`
+	Type string `json:"type,omitempty"`
+	Size int64  `json:"size,omitempty"`
```
Comment (binwiederhier) [inline client/client.go:73]: I don't see anywhere in the code where Size or Type are filled/populated. Or am I missing something? I would have made icon just a string.

**Review by binwiederhier — CHANGES_REQUESTED** [review]

**Comment by wunter8** [comment]
The tests all pass just fine on my computer. I don't think my code affects `TestServer_PublishTooRequests_ShortReplenish` at all

**Review by binwiederhier — APPROVED** [review]
This looks fantastic.

**Pull request merged**